controllers/spacerequest/spacerequest_controller_test.go

```diff
@@ -25,43 +27,907 @@ import (
+				DoesNotExist() // subSpace is gone
+
+			t.Run("spaceRequest is deleted when subSpace is gone", func(t *testing.T) {
+				// when
+				_, err := ctrl.Reconcile(context.TODO(), requestFor(sr))
+
+				// then
+				// space request was deleted
+				require.NoError(t, err)
+				spacerequesttest.AssertThatSpaceRequest(t, srNamespace.Name, sr.GetName(), member1.Client).
+					DoesNotExist() // spaceRequest is gone
+			})
+		})
+
+		t.Run("subSpace is being deleted", func(t *testing.T) {
+			// given
+			srNamespace := newNamespace("jane")
+			parentSpace := spacetest.NewSpace("jane",
+				spacetest.WithCondition(spacetest.Ready()))
+			sr := spacerequesttest.NewSpaceRequest("jane",
+				srNamespace.GetName(),
+				spacerequesttest.WithTierName("appstudio"),
+				spacerequesttest.WithTargetClusterRoles(srClusterRoles),
+				spacerequesttest.WithDeletionTimestamp(), // spaceRequest was deleted
+				spacerequesttest.WithFinalizer())         // has finalizer still
+			subSpace := spacetest.NewSpace("jane-subs",
+				spacetest.WithLabel(toolchainv1alpha1.SpaceRequestLabelKey, sr.GetName()),
+				spacetest.WithLabel(toolchainv1alpha1.SpaceRequestNamespaceLabelKey, sr.GetNamespace()),
+				spacetest.WithSpecTargetCluster("member-1"),
+				spacetest.WithDeletionTimestamp(), // subSpace is being deleted
+				spacetest.WithCondition(spacetest.Ready()),
+				spacetest.WithSpecParentSpace("jane"),
+				spacetest.WithSpecTargetClusterRoles(srClusterRoles),
+				spacetest.WithTierName(sr.Spec.TierName))
+			member1Client := test.NewFakeClient(t, sr, srNamespace)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, appstudioTier, parentSpace, subSpace)
+			ctrl := newReconciler(hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sr))
+
+			// then
+			require.NoError(t, err)
+			spacerequesttest.AssertThatSpaceRequest(t, srNamespace.Name, sr.GetName(), member1.Client).
+				Exists().HasConditions(spacetest.Terminating()) // spaceRequest is in terminating state
+		})
+
+		t.Run("finalizer was already removed", func(t *testing.T) {
+			// given
+			// space request has no finalizer
+			sr := spacerequesttest.NewSpaceRequest("jane",
+				"jane-tenant",
+				spacerequesttest.WithTierName("appstudio"),
+				spacerequesttest.WithTargetClusterRoles(srClusterRoles),
+				spacerequesttest.WithDeletionTimestamp()) // spaceRequest was deleted
+			member1Client := test.NewFakeClient(t, sr, srNamespace)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, appstudioTier)
+			ctrl := newReconciler(hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sr))
+
+			// then
+			// space request is gone
+			require.NoError(t, err)
+			spacerequesttest.AssertThatSpaceRequest(t, srNamespace.Name, "jane-tenant", member1.Client).
+				DoesNotExist()
+		})
```

Comment:
you can drop this test - it doesn't actually verify anything. The fake client automatically removes resources that have deletion timestamp set and don't have any finalizer. The SR is gone right after you add it to the client

Reply:
This is covering this check in the controller code https://github.com/mfrancisc/host-operator/blob/443103fe9267c36c433c15299215e5f7d19ef366/controllers/spacerequest/spacerequest_controller.go#L327-L330

But I understand that it's mainly for coverage and not actually checking anything, if we don't mind to much about the coverage dropping a bit I can remove it.

Reply:
yeah, I would just drop it. No worries about the coverage if it doesn't make any sense to test it :wink: